pkg/controllers/user/alert/watcher/pod.go

```diff
@@ -156,86 +156,82 @@ func (w *PodWatcher) watchRule() error {
-				projectDisplayName := common.GetProjectDisplayName(alert.Spec.ProjectName, w.projectLister)
-
-				data := map[string]string{}
-				data["rule_id"] = ruleID
-				data["group_id"] = alert.Spec.GroupName
-				data["alert_name"] = alert.Spec.DisplayName
-				data["alert_type"] = "podRestarts"
-				data["severity"] = alert.Spec.Severity
-				data["cluster_name"] = clusterDisplayName
-				data["project_name"] = projectDisplayName
-				data["namespace"] = pod.Namespace
-				data["pod_name"] = pod.Name
-				data["container_name"] = containerStatus.Name
-				data["restart_times"] = strconv.Itoa(alert.Spec.PodRule.RestartTimes)
-				data["restart_interval"] = strconv.Itoa(alert.Spec.PodRule.RestartIntervalSeconds)
-
-				if details != "" {
-					data["logs"] = details
-				}
+			clusterDisplayName := common.GetClusterDisplayName(w.clusterName, w.clusterLister)
+			projectDisplayName := common.GetProjectDisplayName(alert.Spec.ProjectName, w.projectLister)
 
-				workloadName, err := w.getWorkloadInfo(pod)
-				if err != nil {
-					logrus.Warnf("Failed to get workload info for %s:%s %v", pod.Namespace, pod.Name, err)
-				}
-				if workloadName != "" {
-					data["workload_name"] = workloadName
-				}
+			data := map[string]string{}
+			data["rule_id"] = ruleID
+			data["group_id"] = alert.Spec.GroupName
+			data["alert_name"] = alert.Spec.DisplayName
+			data["alert_type"] = "podRestarts"
+			data["severity"] = alert.Spec.Severity
+			data["cluster_name"] = clusterDisplayName
+			data["project_name"] = projectDisplayName
+			data["namespace"] = pod.Namespace
+			data["pod_name"] = pod.Name
+			data["container_name"] = containerStatus.Name
+			data["restart_times"] = strconv.Itoa(alert.Spec.PodRule.RestartTimes)
+			data["restart_interval"] = strconv.Itoa(alert.Spec.PodRule.RestartIntervalSeconds)
 
-				if err := w.alertManager.SendAlert(data); err != nil {
-					logrus.Debugf("Error occurred while getting pod %s: %v", alert.Spec.PodRule.PodName, err)
-				}
+			if details != "" {
+				data["logs"] = details
 			}
 
-			return
+			workloadName, err := w.getWorkloadInfo(pod)
+			if err != nil {
+				logrus.Warnf("Failed to get workload info for %s:%s %v", pod.Namespace, pod.Name, err)
+			}
+			if workloadName != "" {
+				data["workload_name"] = workloadName
+			}
+
+			if err := w.alertManager.SendAlert(data); err != nil {
+				logrus.Debugf("Error occurred while getting pod %s: %v", alert.Spec.PodRule.PodName, err)
+			}
 		}
+
+		return
 	}
 
 }
 
 func (w *PodWatcher) getRestartTimeFromTrack(alert *v3.ProjectAlertRule, curCount int32) int32 {
 	name := alert.Name
 	namespace := alert.Namespace
+	now := time.Now()
+	currentRestartTrack := restartTrack{Count: curCount, Time: now}
+	currentRestartTrackArr := []restartTrack{currentRestartTrack}
 
 	obj, ok := w.podRestartTrack.Load(namespace + ":" + name)
 	if !ok {
+		w.podRestartTrack.Store(namespace+":"+name, currentRestartTrackArr)
```

Comment:
Do we really need to store the whole track in the map? I think you just need the latest one. So maybe we just need to store that.